subprojects/core/src/main/java/org/gradle/caching/internal/tasks/DefaultTaskOutputCachingBuildCacheKeyBuilder.java

```diff
@@ -140,7 +154,10 @@ public boolean isValid() {
 
         @Override
         public String toString() {
-            return String.valueOf(hashCode);
```

Comment:
Using toString() for this is a bit loose. If it's effectively part of the BuildCacheKey for a key to be able to describe what it represents, it should be a formal method on that interface.